sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/catalog/ExternalCatalogEventSuite.scala

```diff
@@ -209,4 +209,67 @@ class ExternalCatalogEventSuite extends SparkFunSuite {
     catalog.dropFunction("db5", "fn4")
     checkEvents(DropFunctionPreEvent("db5", "fn4") :: DropFunctionEvent("db5", "fn4") :: Nil)
   }
+
+  testWithCatalog("partition") { (catalog, checkEvents) =>
+    val path1 = Files.createTempDirectory("db_")
+    val path2 = Files.createTempDirectory(path1, "tbl_")
+    val uri1 = preparePath(path1)
+    val uri2 = preparePath(path2)
+
+    // CREATE
+    val dbDefinition = createDbDefinition(uri1)
+
+    val storage = CatalogStorageFormat.empty.copy(
+      locationUri = Option(uri2))
+
+    val tableDefinition = CatalogTable(
+      identifier = TableIdentifier("tbl1", Some("db5")),
+      tableType = CatalogTableType.MANAGED,
+      storage = storage,
+      schema = new StructType().add("id", "long"))
+
+    val partition = CatalogTablePartition(spec = Map("key1" -> "1", "key2" -> "2"),
+      storage = CatalogStorageFormat.empty)
+
+    catalog.createDatabase(dbDefinition, ignoreIfExists = false)
+    checkEvents(CreateDatabasePreEvent("db5") :: CreateDatabaseEvent("db5") :: Nil)
+
+    catalog.createTable(tableDefinition, ignoreIfExists = false)
+    checkEvents(CreateTablePreEvent("db5", "tbl1") :: CreateTableEvent("db5", "tbl1") :: Nil)
+
+    catalog.createPartitions("db5", "tbl1", Seq(partition), ignoreIfExists = false)
+    checkEvents(CreatePartitionPreEvent("db5", "tbl1", Seq(partition)) ::
+      CreatePartitionEvent("db5", "tbl1", Seq(partition)) :: Nil)
+
+    catalog.createPartitions("db5", "tbl1", Seq(partition), ignoreIfExists = true)
+    checkEvents(CreatePartitionPreEvent("db5", "tbl1", Seq(partition)) ::
+      CreatePartitionEvent("db5", "tbl1", Seq(partition)) :: Nil)
```

Comment:
Ur, is this correct? We already created the same `partition` at line 240 and we tried with `ignoreIfExists=true` here. I guess `CreatePartitionEvent` should not occur.

Reply:
I agree with you, it looks incorrect, but we can not check if the partition has been created or not created in case of using  `ignoreIfExists=true`

Reply:
If then, this is not helpful in that case because we cannot distinguish the difference.